x-pack/solutions/observability/plugins/slo/server/lib/rules/slo_burn_rate/executor.ts

```diff
@@ -84,7 +85,17 @@ export const getRuleExecutor = (basePath: IBasePath) =>
     }
 
     const sloRepository = new KibanaSavedObjectsSLORepository(soClient, logger);
-    const slo = await sloRepository.findById(params.sloId);
+    let slo: SLODefinition;
+    try {
+      slo = await sloRepository.findById(params.sloId);
+    } catch (err) {
+      throw createTaskRunError(
+        new Error(
+          `Rule "${options.rule.name}" ${options.rule.id} is referencing an SLO which cannot be found: "${params.sloId}": ${err.message}`
```

Comment:
Technically the repository can also throw when the SLO is invalid (not only not found), but both categorize as User error anyway. Maybe we can tweak the message to include this variant, without handling both case explicitly.

Reply:
I'll let the message get tweaked in another PR, in hopes of merging this in time for Monday's upgrade ...